### DISCUSSION TITLE
Build or rebuild binary target when tests are run

### DISCUSSION THREAD

**Discussion by andy5995** [mesonbuild/meson#11550] [Q&A] [answered]
For script-based tests I need meson to build or rebuild the binary.

The binary target is created in the [top-level meson.build](https://github.com/theimpossibleastronaut/rmw/blob/master/meson.build). The script-based tests are in [tests/meson.build](https://github.com/theimpossibleastronaut/rmw/blob/master/test/meson.build).

Is there a way to build the binary target when `meson test` is run?

**Comment by andy5995** [comment]
This is a case where the answer is in the docs.

I add a test setup for the main binary target and set parallelism to `false`

    test('build binary', main_bin, is_parallel : false)

https://mesonbuild.com/Unit-tests.html#test-parameters

https://mesonbuild.com/Unit-tests.html#parallelism

And the fix magically appears at:

https://github.com/theimpossibleastronaut/rmw/pull/396

**Comment by eli-schwartz** [comment, marked as answer]
Pretty sure what you want here is `depends: main_bin`, see https://mesonbuild.com/Reference-manual_functions.html#test

> specifies that this test depends on the specified target(s), even though it does not take any of them as a command line argument. This is meant for cases where test finds those targets internally, e.g. plugins or globbing.

By adding a test that executes main_bin without args and expects it to return success, all you do is make the entire test suite depend on that bin before running *any* tests, but `depends:` does the same job except better -- it also works when you run

```
meson test single_test_name
```

**Reply by andy5995** [reply]
Yeah, that's what I want. Thanks @eli-schwartz
